### PR TITLE
Fix special_return method's result wrapping to be more dynamic

### PR DESCRIPTION
**Pull request by tscholz**
Hi,

there is a little issue with the FileList SPECIAL_RETURNs:
```
class CustomFileList < Rake::FileList
end

c = CustomFileList.new #=> []
c.class #=> CustomFileList

c.uniq.class #=> Rake::FileList but should be a CustomFileList
```
What do you think about this fix?

### CODE REVIEW

**Comment by hsbt** [comment]
@drbrain Can you review this?

**Comment by tscholz** [comment]
Hi,

any news on this?

**Comment by tscholz** [comment]
@hsbt, @drbrain

Hi,
I'd like to kindly ask if someone could have a look at this? I know this is not a very important issue and I'm sure there are a lot of other things to do. Nevertheless it would be nice to see some progress on this.

File: lib/rake/ext/time.rb [hunk 1]
```diff
@@ -4,13 +4,15 @@
...
 require 'rake/late_time'
 
-class Time # :nodoc: all
-  alias rake_original_time_compare :<=>
-  def <=>(other)
-    if Rake::EarlyTime === other || Rake::LateTime === other
-      - other.<=>(self)
-    else
-      rake_original_time_compare(other)
+if RUBY_VERSION < "1.9"
```
Comment (hsbt) [inline lib/rake/ext/time.rb]: Is it necessary for your issue?

**Comment by hsbt** [comment]
@tscholz It strange behavior for me. Your proposal is reasonable. I will merge after your feedback on my comment.

File: lib/rake/ext/time.rb (hunk 1 above)
Comment (hsbt) [inline lib/rake/ext/time.rb]: this change already merged in master.

**Pull request merged**